src/js/angular/repositories/ontop-repo.directive.js

```diff
@@ -0,0 +1,300 @@
+                Upload.upload({
+                    url: 'rest/repositories/uploadFile',
+                    data: {uploadFile: $scope.uploadFile}
+                })
+                    .success(function (data) {
+                        if (!data.success) {
+                            toastr.error(data.errorMessage);
+                        } else {
+                            $scope.ontopRepoFileNames[param] = $scope.uploadFile.name;
+                            $scope.repositoryInfo.params[param].value = data.fileLocation;
+                        }
+                        $scope.uploadFileLoader = false;
+                    }).error(function (data) {
+                    const msg = getError(data);
+                    toastr.error(msg, 'Error');
+                    $scope.uploadFile = '';
+                    $scope.uploadFileLoader = false;
+                });
+            }
+        }
+
+        $scope.concatURL = function(labelName) {
+            if (labelName === 'hostName' || labelName === 'port' || labelName === 'databaseName') {
+                let result = $scope.selectedDriver.urlTemplate;
+                if ($scope.selectedDriver.jdbc.hostName) {
+                    if ($scope.selectedDriver.jdbc.port) {
+                        result = result.replace('{hostport}',
+                            $scope.selectedDriver.jdbc.hostName + ':' + $scope.selectedDriver.jdbc.port);
+                    } else {
+                        result = result.replace('{hostport}', $scope.selectedDriver.jdbc.hostName);
+                    }
+                }
+
+                if ($scope.selectedDriver.jdbc.databaseName) {
+                    result = result.replace('{database}', $scope.selectedDriver.jdbc.databaseName);
+                }
+
+                $scope.selectedDriver.jdbc.url = result;
+            }
+        }
+
+        $scope.getInputType = function (labelName) {
+            switch (labelName) {
+                case 'password':
+                    return 'password';
+                case 'port':
+                    return 'number';
+                default:
+                    return 'text';
+            }
+        }
+
+        $scope.checkForRequiredOntopFiles = function () {
+            // Should guarantee that code will be executed in sequential manner,
+            // because properties file is not created yet
+            return updateProperties()
+                .then(function () {
+                    const missingRequired = REQUIRED_ONTOP_REPO_PARAMS.filter(function (requiredFile) {
+                        return !$scope.repositoryInfo.params[requiredFile].value;
+                    });
+                    if (missingRequired.length > 0) {
+                        toastr.error('Missing required ontop repo file');
+                        return Promise.reject('Missing required ontop repo file');
+                    }
+                }).catch(function (err) {
```

Comment:
This catch is useless here. The error will be propagated eiter way if you don't catch it here.